### PR TITLE
Add-on Store: Add filter by channel

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Part of #13985 

### Summary of the issue:
A user may want to browse add-ons by different channels.

A bug was introduced to the add-on store which caused only one channel version per add-on to be available in the listing.
A user may wish to browse add-ons by channel, as well as view all channels for an add-on.

### Description of user facing changes
Added "channel" to the listing columns.
Added a filter by channel.
A bug was fixed where only one channel version per add-on was listed when listing all available add-ons for all channels.

### Description of development approach
Add-ons are now collected by channel and add-on ID.
A channel filter was added similar to the status filter.

### Testing strategy:

1. Open the add-on store
1. Jump to the filter-by channel field (`alt+n`)
1. Filter by each channel grouping, ensure expected add-ons are displayed.

### Known issues with pull request:
None
### Change log entries:
N/A

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by seanbudd** [comment]
A different accelerator key is need for the channel filter, alt+c should close the dialog.

**Comment by josephsl** [comment]
Hi, how about Alt+N or Alt+L? Thanks.

**Comment by XLTechie** [comment]
@josephsl wrote:
> Hi, how about Alt+N or Alt+L? Thanks.

I like Alt+l, personally, but either one seems reasonable.

**Review by michaelDCurran — APPROVED** [review]

File: tests/manual/addonStore.md [hunk 1]
```diff
@@ -12,6 +12,13 @@ Add-ons can be filtered by status.
     - Disabled: should include manually disabled add-ons and add-ons disabled due to incompatibility
     - Available: should include add-ons available for install
 
+### Filter add-ons by channel
+Add-ons can be filtered by channel: e.g. stable, beta, dev.
+
+1. Open the add-on store
+1. Jump to the filter-by channel field (`alt+c`)
```
Comment (seanbudd) [inline tests/manual/addonStore.md:19]:
```suggestion
1. Jump to the filter-by channel field (`alt+n`)
```

**Pull request merged**